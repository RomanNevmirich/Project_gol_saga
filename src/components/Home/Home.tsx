import React from 'react';
import { Link } from "react-router-dom";

//Статичный компонент с текстом
export const Home = () => (
      <>
        <nav>
            <Link to="/about">Правила игры</Link>
            <p/>
            <Link to="/game">Перейти к игре</Link>
        </nav>
        <main>
            <h3>Project description</h3>

            <h4>Цель:</h4>
            <span>Реализовать готовое для продакшн приложение</span>
            <h4>Описание/Пошаговая инструкция выполнения домашнего задания:</h4>
            <span>1. конфигурируем приложение самостоятельно через webpack + babel</span><br/>
            <span>2. разработка компонентов ведется отдельно в библиотеке компонентов</span><br/>
            <span>3. четко прослеживаемая архитектура</span><br/>
            <span>4. saga для управления всеми side effects</span><br/>
            <span>5. устойчиво к изменениям за счет тестов</span><br/>
            <span>6. не менее 50% прокрытие тестами</span><br/>
            <span>7. оценка качества через stryker должна быть на высоком уровне</span><br/>
            <span>8. полная работоспособность и решение конкретной задачи</span><br/>
            <span>9. минимально сконфигурированный и работоспособный SSR</span><br/>
            <h4>Критерии оценки:</h4>
            <span>1. правильная конфигурация</span><br/>
            <span>2. хорошая архитектура</span><br/>
            <span>3. хорошо проработанная библиотека компонентов</span><br/>
            <span>4. хорошие тест кейсы</span><br/>
            <span>5. работоспособность и решение поставленной задачи</span><br/>
            <span>6. SSR</span><br/>
        </main>
      </>
);