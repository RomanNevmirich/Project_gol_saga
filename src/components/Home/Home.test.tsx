import React from 'react';
import { render } from '@testing-library/react';
import { Home } from './Home';
import { HashRouter } from "react-router-dom";

describe('Home component check', () => {
  
  it('Home render', () => {
    const { asFragment } = render(<HashRouter><Home /></HashRouter>);

    expect(asFragment()).toMatchSnapshot();
  });

});