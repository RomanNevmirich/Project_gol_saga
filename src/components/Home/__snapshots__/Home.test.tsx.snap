// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component check Home render 1`] = `
<DocumentFragment>
  <nav>
    <a
      href="#/about"
    >
      Правила игры
    </a>
    <p />
    <a
      href="#/game"
    >
      Перейти к игре
    </a>
  </nav>
  <main>
    <h3>
      Project description
    </h3>
    <h4>
      Цель:
    </h4>
    <span>
      Реализовать готовое для продакшн приложение
    </span>
    <h4>
      Описание/Пошаговая инструкция выполнения домашнего задания:
    </h4>
    <span>
      1. конфигурируем приложение самостоятельно через webpack + babel
    </span>
    <br />
    <span>
      2. разработка компонентов ведется отдельно в библиотеке компонентов
    </span>
    <br />
    <span>
      3. четко прослеживаемая архитектура
    </span>
    <br />
    <span>
      4. saga для управления всеми side effects
    </span>
    <br />
    <span>
      5. устойчиво к изменениям за счет тестов
    </span>
    <br />
    <span>
      6. не менее 50% прокрытие тестами
    </span>
    <br />
    <span>
      7. оценка качества через stryker должна быть на высоком уровне
    </span>
    <br />
    <span>
      8. полная работоспособность и решение конкретной задачи
    </span>
    <br />
    <span>
      9. минимально сконфигурированный и работоспособный SSR
    </span>
    <br />
    <h4>
      Критерии оценки:
    </h4>
    <span>
      1. правильная конфигурация
    </span>
    <br />
    <span>
      2. хорошая архитектура
    </span>
    <br />
    <span>
      3. хорошо проработанная библиотека компонентов
    </span>
    <br />
    <span>
      4. хорошие тест кейсы
    </span>
    <br />
    <span>
      5. работоспособность и решение поставленной задачи
    </span>
    <br />
    <span>
      6. SSR
    </span>
    <br />
  </main>
</DocumentFragment>
`;
